src/MoonSharp.Interpreter/Interop/StandardDescriptors/ReflectionMemberDescriptors/MethodMemberDescriptorCoroutine.cs

```diff
@@ -0,0 +1,48 @@
+using System.Reflection;
+
+namespace MoonSharp.Interpreter.Interop 
+{
+    public class MethodMemberDescriptorCoroutine : MethodMemberDescriptor 
+	{
+        public MethodMemberDescriptorCoroutine(MethodBase methodBase, InteropAccessMode accessMode = InteropAccessMode.Default) : base(methodBase, accessMode) 
+		{
+        }
+
+        public override DynValue GetValue(Script script, object obj) 
+		{
+            var enumerateYielder = script.DoString(@"return function (callable) 
+    return function (...)
+        for y in callable(...) do
+            if coroutine.is_return_value(y) then
+                return coroutine.get_return_value(y)
+            else 
+                coroutine.yield(y)
+            end
+        end
+    end
+end", null, MethodInfo + "_yielder");
```

Comment:
Does it make sense to parse this lua code one for performance reasons? Could this lua function be pulled out and kept as a global function to be used when needed? I've done something similar in my own code, though all on the lua side without the C# interconnection work this PR does to make it nice and clean.